### PR TITLE
fix: enum values should honor JsonProperty annotations

### LINKED ISSUE #3411: Allow to specify POJO enum values for the corresponding generated CRD enum
When a POJO contains an enum, I would like to serialize it into the corresponding CRD with a specific name instead of the field one.
For example, if an enum is defined this way:

```java
public enum MyEnum {
       Foo,
       Bar,
```

The corresponding CRD will be the following:

```yaml
name:
    enum:
    - Foo
    - Bar
    type: string
```

while I would like to have:

```yaml
name:
    enum:
    - foo
    - bar
    type: string
```

or anyway any other name (i.e. if for any crazy reason I want `Foo` to be `aaa` in the CRD).
I tried the `JsonProperty("foo")` on the enum but with no luck.

### LINKED ISSUE #3431: CRD generator doesn't properly add renamed properties to `required` list
The CRD generator currently doesn't account for renamed properties when checking for required field: the field will always be added with the original name without taking any potential overrides.

### PR DESCRIPTION
**Pull request by metacosm**
Fixes #3411
FIxes #3431 

## Type of change

 - [x] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] Code contributed by me aligns with current project license: [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0)
 - [x] I Added [CHANGELOG](../CHANGELOG.md) entry regarding this change
 - [x] I have implemented unit tests to cover my changes
 - [x] I have added/updated the [javadocs](https://www.javadoc.io/doc/io.fabric8/kubernetes-client/latest/index.html) and other [documentation](https://github.com/fabric8io/kubernetes-client/blob/master/doc/CHEATSHEET.md) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=fabric8io_kubernetes-client) report
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

File: crd-generator/api/src/main/java/io/fabric8/crd/generator/AbstractJsonSchema.java [hunk 1]
```diff
@@ -132,85 +120,208 @@ protected T internalFrom(TypeDef definition, String... ignore) {
...
 
-  private void updatePropertyFromAnnotationIfNeeded(Property property, String name, Property[] updated,
-    AnnotationRef a) {
-    if (updated[0] != null) {
-      LOGGER.debug("Property {} has already been renamed to {}", name, updated[0].getName());
-      return;
+  private Map<String, Method> indexPotentialAccessors(TypeDef definition) {
+    final List<Method> methods = definition.getMethods();
+    final Map<String, Method> accessors = new HashMap<>(methods.size());
+    methods.stream()
```
Comment (iocanel) [inline crd-generator/api/src/main/java/io/fabric8/crd/generator/AbstractJsonSchema.java:157]: what about inherited properties and accessors?

**Review by iocanel — COMMENTED** [review]
Overall, looks good. The only thing I missed is what happens with inherited types.

File: crd-generator/api/src/main/java/io/fabric8/crd/generator/AbstractJsonSchema.java (hunk 1 above)
Comment (metacosm) [reply crd-generator/api/src/main/java/io/fabric8/crd/generator/AbstractJsonSchema.java:157]: Aren't the TypeDef that are passed there "complete"? Plus, this is not different from what was done before i.e. the previous implementation was only checking the TypeDef's methods for accessors. This bit only makes it slightly more efficient by avoiding to iterate over all methods for each property.

**Review by manusa — APPROVED** [review]

File: crd-generator/api/src/main/java/io/fabric8/crd/generator/AbstractJsonSchema.java (hunk 1 above)
Comment (metacosm) [reply crd-generator/api/src/main/java/io/fabric8/crd/generator/AbstractJsonSchema.java:157]: @iocanel if you think this is an issue, I think we should open a different issue.

**Review by rohanKanojia — APPROVED** [review]

File: crd-generator/api/src/main/java/io/fabric8/crd/generator/AbstractJsonSchema.java (hunk 1 above)
Comment (iocanel) [reply crd-generator/api/src/main/java/io/fabric8/crd/generator/AbstractJsonSchema.java:157]: `TypeDef` instances do not include properties and method available in super classes. This is why I recently added `RichTypeDef` that allows accessing to superclass properties and methods.

As long as we are aware of this, we can proceed and cover this in  a future pr.

**Review by iocanel — APPROVED** [review]

**Pull request merged**